job/job.go

```diff
@@ -74,44 +77,53 @@ func HasActiveWorkflow(hwID packet.HardwareID) (bool, error) {
 }
 
 // CreateFromDHCP looks up hardware using the MAC from cacher to create a job
-func CreateFromDHCP(mac net.HardwareAddr, giaddr net.IP, circuitID string) (Job, error) {
+func CreateFromDHCP(ctx context.Context, mac net.HardwareAddr, giaddr net.IP, circuitID string) (Job, error) {
 	j := Job{
 		mac:   mac,
 		start: time.Now(),
 	}
 
-	d, err := discoverHardwareFromDHCP(mac, giaddr, circuitID)
+	span := trace.SpanFromContext(ctx)
```

Comment:
Why the heck didn't I think of that!? Thanks!

c7fa1aa1816cd052ba37bee96a0515f6d90707c1

Reply:
After replying to your later content I wonder if it's even right to end that span since it comes from ctx. It hasn't broken anything so we can let it stand for now I think.

Reply:
Yeah, ok. I removed those span.Ends entirely. They shouldn't have been there since it's a context span.

Reply:
40b8c54bdf032b8edba61db6d2e1e2979a7f9ac9